service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/gcsbucket/ControlledGcsBucketResource.java

```diff
@@ -91,6 +92,11 @@ private ControlledGcsBucketResource(ControlledResourceFields common, String buck
     validate();
   }
 
+  public ControlledGcsBucketResource(DbResource dbResource, String bucketName) {
```

Comment:
Maybe comment that this constructor is only used for building the resource from the database and not validating the bucket name.